ipatests/test_xmlrpc/test_host_plugin.py

```diff
@@ -337,6 +341,84 @@ def test_add_host_with_null_password(self, host):
         host.track_create()
         host.check_create(result)
 
+    @staticmethod
+    def modify_host_config(host_tracker, value):
+        command = host_tracker.make_command(
+            'config_mod',
+            **dict(
+                setattr=u'maxhostnamelen={}'.format(value)))
+        command()
+
+    def test_lower_hostname_limit_below_maxhostnamelimit(self):
+        """Change hostnamelimit value below default. Should fail"""
+        testhost = HostTracker(name=u'testshost')
+        with raises_exact(errors.ValidationError()):
+            self.modify_host_config(testhost, MAXHOSTNAMELEN // 2)
+
+    def test_raise_hostname_limit_above_maxhostnamelimit(self):
+        """Raise hostnamelimit above default. Should pass"""
+        testhost = HostTracker(name=u'testshost',
+                               fqdn=u'{label}.{label}.{label}'.format(
```

Comment:
According to the test comment, it should only try to set ipa config-mod maxhostname to a value > MAXHOSTNAMELEN. No need to call host_create_command here.